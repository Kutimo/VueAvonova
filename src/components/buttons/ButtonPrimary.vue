<script lang="ts">
export default {
  name: "ButtonPrimary",
  props: {
    buttonText: {
      type: String,
      required: true,
    },
    icon: {
      type: String,
      default: "",
    },
    to: {
      type: [String, Object],
      default: "",
    },
  },
};
</script>

<template>
  <a
    tabindex="0"
    class="w-fit cursor-pointer select-none rounded-28 bg-green-1200 px-28 py-10 font-body text-base font-medium leading-5 text-green-300 outline-none ring-black ring-offset-2 transition-all duration-500 hover:bg-green-600 hover:text-black focus-visible:ring-2 active:bg-green-400"
  >
    <span v-if="icon" class="icon" v-html="icon"></span>
    {{ buttonText }}
  </a>
</template>
