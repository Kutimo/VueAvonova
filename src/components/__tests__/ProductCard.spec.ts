import { describe, it, expect, vi } from "vitest";
import { mount } from "@vue/test-utils";
import CardComponent from "../cards/ProductCard.vue";
import ButtonPrimary from "../buttons/ButtonPrimary.vue";

describe("CardComponent", () => {
  it("renders card with correct props and buttons", async () => {
    const cardIcon = "example.png";
    const cardHeader = "Example Header";
    const cardContent = "Example Content";

    const wrapper = mount(CardComponent, {
      props: {
        cardIcon,
        cardHeader,
        cardContent,
      },
      global: {
        components: {
          ButtonPrimary,
        },
      },
    });

    // Check if props are rendered correctly
    expect(wrapper.find("img").exists()).toBe(true);
    expect(wrapper.find("img").attributes("src")).toBe(cardIcon);
    expect(wrapper.find("h4").text()).toBe(cardHeader);
    expect(wrapper.find("p").text()).toBe(cardContent);

    // Check if buttons are rendered and emit correct events
    const buttons = wrapper.findAllComponents(ButtonPrimary);
    expect(buttons.length).toBe(2);

    // Simulate button clicks and check if corresponding methods are called
    await buttons[0].trigger("click");
    await buttons[1].trigger("click");
    // TODO: Check if methods are called
    // expect(wrapper.emitted().onReadMore).toBeTruthy();
    // expect(wrapper.emitted().onBookAppointment).toBeTruthy();
  });
});
