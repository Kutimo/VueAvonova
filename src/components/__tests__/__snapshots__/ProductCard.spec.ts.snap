// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > renders the component 1`] = `
"<div class="m-10 flex w-[250px] flex-col items-center overflow-hidden rounded-4 bg-green-600 p-4 shadow-md">
  <!--v-if-->
  <!--v-if-->
  <p class="body-md px-20 text-center"></p>
  <div class="flex w-full justify-around p-4"><a to="handleClick" tabindex="0" class="cursor-pointer select-none rounded-28 bg-green-1200 px-24 py-12 font-body text-base font-medium leading-5 text-green-300 outline-none ring-black ring-offset-2 transition-all duration-500 hover:bg-green-600 hover:text-black focus-visible:ring-2 active:bg-green-400">
      <!--v-if--> Les mer
    </a><a to="handleClick" tabindex="0" class="cursor-pointer select-none rounded-28 bg-green-1200 px-24 py-12 font-body text-base font-medium leading-5 text-green-300 outline-none ring-black ring-offset-2 transition-all duration-500 hover:bg-green-600 hover:text-black focus-visible:ring-2 active:bg-green-400">
      <!--v-if--> Bestill
    </a></div>
</div>"
`;
